File: 024.rb
# What is the millionth lexicographic permutation of the digits 
# 0, 1, 2, 3, 4, 5, 6, 7, 8 and 9?

digits = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
permutations = Array.new
digits.permutation.to_a.map { |perm| permutations.push(perm.join) }
puts permutations.sort![999999]

File: clojure/src/project_euler_clojure/024.clj
;; What is the millionth lexicographic permutation of the digits
;; 0, 1, 2, 3, 4, 5, 6, 7, 8 and 9?

(ns project-euler-clojure.024
  (:require [clojure.math.combinatorics :as combo]))

(defn problem-024 []
  (->> (combo/permutations (range 10))
       (#(nth % 999999))
       (clojure.string/join)
       (bigint)))
